### PR TITLE
Improve dev docs for `ActionType`

### PR DESCRIPTION
**Pull request by DaveCTurner**
The semantics of `ActionType` are somewhat confusing; this commit
expands its Javadocs a bit to try and clarify how to use it.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/ActionType.java [hunk 1]
```diff
@@ -27,8 +49,17 @@ public static ActionType<ActionResponse.Empty> emptyResponse(String name) {
...
+     * (using a client obtained from {@link Client#getRemoteClusterClient}). If the action is only to be executed on the local cluster then
+     * declare it using {@link #localOnly} instead.
+     * <p>
+     * There is no facility for directly executing an action on a different node in the local cluster. To achieve this, implement an action
+     * which runs on the local node and knows how to use the {@link TransportService} to forward the request to a different node. There are
+     * several utilities that help implement such an action, including {@link TransportNodesAction} or {@link TransportMasterNodeAction}.
+     *
+     * @param name           The name of the action, which must be unique across actions. When executed on a remote cluster, this is the
+     *                       ID of the transport action which is sent to the handling node in the remote cluster.
+     * @param responseReader A reader for the response type, which is only used by actions executed on a remote cluster.
```
Comment (joegallo) [inline server/src/main/java/org/elasticsearch/action/ActionType.java:62]: > A reader for the response type, which is only used by actions executed on a remote cluster.

Is that a reader for the response from the action executed on the remote cluster? I'm trying to read the as-written words incorrectly and I feel like I can convince myself that those words say that it's the action executed on the remote cluster that uses the reader.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/ActionType.java:62]: Fair, I see the ambiguity. Let me rearrange the words, sec.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/ActionType.java]: 1b8526bb9659 any better?

Comment (joegallo) [reply server/src/main/java/org/elasticsearch/action/ActionType.java]: SO MUCH BETTER. Thank you! I love it!

**Review by joegallo — APPROVED** [review]
❤️

**Review by ldematte — APPROVED** [review]
Late to the party, but I really like it! It would have saved me some time while I was trying to understand action registration/execution.

**Pull request merged**